my-code/src/Contexts/MoviesData.tsx

```diff
@@ -0,0 +1,95 @@
+import axios from "axios";
+import { createContext, ReactNode, useContext, useEffect, useState } from "react";
+
+interface MoviesDataProviderProps {
+    children: ReactNode;
+}
+
+interface MoviesDataContextProps {
+    movies: MovieProps[];
+    getMoviesData: (search: string) => Promise<void>;
+    favouriteMovies: FavouriteMoviesProps[];
+    addToFavourites: (movieID: string) => void;
+    isFavourite: (movieID: string) => boolean;
+    isLoading: boolean;
+}
+
+interface MovieProps {
+    Actors: string;
+    Awards: string;
+    Country: string;
+    Director: string;
+    Genre: string;
+    Language: string;
+    Metascore: string;
+    Plot: string;
+    Poster: string;
+    Released: string;
+    Title: string;
+    Year: string;
+    imdbRating: string;
+    imdbID: string;
+}
+
+interface FavouriteMoviesProps {
+    imdbID: string;
+}
+
+const MoviesDataContext = createContext({} as MoviesDataContextProps)
+
+export function MoviesDataProvider({ children }: MoviesDataProviderProps) {
+
+    const [movies, setMovies] = useState<MovieProps[]>([])
+    const [favouriteMovies, setFavouriteMovies] = useState<FavouriteMoviesProps[]>(
+        JSON.parse(localStorage.getItem('favouriteMovies') || '[]')
+    )
+    const [isLoading, setIsLoading] = useState(false)
+
+    async function getMoviesData(search: string) {
+        if (search.length > 0 && search.length < 3) return;

```

Comment:
There is a movie called pi: https://www.imdb.com/title/tt0138704/?ref_=nv_sr_srsg_8